src/components/ChattingScreen.js

```diff
@@ -1,31 +1,103 @@
-import React from 'react';
-import styled from 'styled-components';
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
 
-import Header from './Header';
-import MessageSender from './MessageSender';
+import Header from "./Header";
+import MessageSender from "./MessageSender";
 
 export default function ChattingScreen() {
-	const EUNKO = 'https://img.techpowerup.org/200908/eun.png';
-	const COOL = 'https://img.techpowerup.org/200908/NjRiY2JjOGU5YzQz.png';
-
-	const MSGLIST = [
-		{ user: true, content: '안녕하세요 12기 프론트엔드 개발자분들' },
-		{ user: true, content: '저희의 대화를 마음껏 조작해보세요 💌' },
-		{ user: true, content: '상단에 프로필을 눌러서 발신자 변경하면 됩니당~' },
-		{ user: false, content: '안녕 은아' },
-		{ user: false, content: '뭐해 ?' },
-		{ user: true, content: '시원아 넌 최고의 팀장이야' },
-		{ user: false, content: '나도 아니까  ' },
-		{ user: false, content: '그만 말해줘도 돼' },
-		{ user: true, content: 'ㅠㅠ' },
-	];
-
-	return (
-		<Wrapper>
-			<Header></Header>
-			<MessageSender></MessageSender>
-		</Wrapper>
-	);
+  const MSGLIST = [
+    { user: true, content: "안녕하세요 12기 프론트엔드 개발자분들" },
+    { user: true, content: "저희의 대화를 마음껏 조작해보세요 💌" },
+    { user: true, content: "상단에 프로필을 눌러서 발신자 변경하면 됩니당~" },
+    { user: false, content: "안녕 은아" },
+    { user: false, content: "뭐해 ?" },
+    { user: true, content: "시원아 넌 최고의 팀장이야" },
+    { user: false, content: "나도 아니까  " },
+    { user: false, content: "그만 말해줘도 돼" },
+    { user: true, content: "ㅠㅠ" },
+  ];
+  const [InputText, setInputText] = useState("");
+  const [MessageList, setMessageList] = useState(MSGLIST);
+  const [User, setUser] = useState(true);
+
+  useEffect(() => {
+    window.scrollBy({
+      botton: 100,
+      behavior: "smooth",
+    });
+  }, [MessageList]);
+
+  const handleInput = (event) => {
+    event.preventDefault();
+
+    setInputText(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    //prevent refresh right after submit
+    event.preventDefault();
+
+    //handle exception when user typed nothing
+    if (InputText === "") {
+      alert("내용을 입력하세요");
+      return;
+    }
+    //concat to previous MessageList
+    const nextMessageList = MessageList.concat({
+      user: User,
+      content: InputText,
+    });
+    //update MessageList
+    setMessageList(nextMessageList);
+    setInputText("");
+  };
+
+  return (
+    <>
+      {/* child component updates state in parent component   */}
+      <Header user={User} onClick={() => setUser(!User)}></Header>
+
+      <Chat>
+        <MessageSender {...MessageList}></MessageSender>
+      </Chat>
+
+      <InputBox>
+        <input value={InputText} onChange={handleInput} />
+        <button onClick={handleSubmit}>전송</button>
+      </InputBox>
+    </>
+  );
 }
 
```

Comment:
이렇게 justify-content에 props로 처리해줘도 간결해지고 좋을 것 같습니다 : )

```suggestion
const MessageRow = styled.div`
	display: flex;
	flex-shrink: 0;
	width: 100%;
	align-items: center;
	justify-content: ${(props) => (props.sending ? 'flex-end' : 'flex-start')};
`;
const Message = styled.div`
	display: flex;
	align-items: center;
	justify-content: center;
	text-align: center;
	background-color: white;
	height: 30px;
	padding: 8px;
	border-radius: 10px;
	box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.1);
`;

const Img = styled.img`
	width: 50px;
	height: 50px;
	border-radius: 20px;
	margin: 10px;
	box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.25);
`;
```

Reply:
와... 정말 뒤통수 한대 얻어 맞은것 같습니당 ㅠㅠ
<Header> 에서와 마찬가지로 중복되는 코드를 줄이고 싶었는데 삼항연산자, styled-component props 를 이용하는 방법이 있었군요 ㅠㅠ 감사합니다!!
추가로 css에서 flex-direction으로 reverse를 할 수 있다는 점, box-shadow를 주면 화면이 더 귀여워진다(?) 는 점도 배웠어요! 감사합니다 😄